transport/controlbuf.go

```diff
@@ -459,28 +459,37 @@ func (l *loopyWriter) incomingSettingsHandler(s *incomingSettings) error {
 
 func (l *loopyWriter) headerHandler(h *headerFrame) error {
 	if l.side == serverSide {
-		if h.endStream { // Case 1.A: Server wants to close stream.
-			// Make sure it's not a trailers only response.
-			if str, ok := l.estdStreams[h.streamID]; ok {
-				if str.state != empty { // either active or waiting on stream quota.
-					// add it str's list of items.
-					str.itl.enqueue(h)
-					return nil
-				}
-			}
-			if err := l.writeHeader(h.streamID, h.endStream, h.hf, h.onWrite); err != nil {
-				return err
+		// Case 1.A: Server got headers from client, a corresoponding stream must be registered with loopy.
+		if h.hf == nil {
```

Comment:
Instead of reusing an existing field and special-casing the absence of that field for this purpose, it would be less error-prone to have an explicit signal that this is a header that came from the client.  Either a new type and handler for it, or another bool/enum to signify this would be preferable IMO.